src/main/java/uk/gov/justice/digital/writer/Writer.java

```diff
@@ -7,84 +7,91 @@
 import org.jetbrains.annotations.NotNull;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import uk.gov.justice.digital.config.JobArguments;
 import uk.gov.justice.digital.converter.dms.DMS_3_4_6;
 import uk.gov.justice.digital.domain.model.SourceReference;
 import uk.gov.justice.digital.exception.DataStorageException;
 import uk.gov.justice.digital.service.DataStorageService;
-import uk.gov.justice.digital.service.SourceReferenceService;
-
-import javax.inject.Inject;
 
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.function.Consumer;
 
-import static org.apache.spark.sql.functions.col;
-import static uk.gov.justice.digital.converter.dms.DMS_3_4_6.Operation.*;
+import static uk.gov.justice.digital.converter.dms.DMS_3_4_6.Operation.getOperation;
 import static uk.gov.justice.digital.converter.dms.DMS_3_4_6.ParsedDataFields.OPERATION;
-import static uk.gov.justice.digital.zone.RawZone.PRIMARY_KEY_NAME;
 
-public class StructuredZoneCDC extends StructuredZone {
+public abstract class Writer {
+
+    private static final Logger logger = LoggerFactory.getLogger(Writer.class);
 
-    private static final Logger logger = LoggerFactory.getLogger(StructuredZone.class);
+    public abstract void writeValidRecords(
+            SparkSession spark,
+            String destinationPath,
+            SourceReference.PrimaryKey primaryKey,
+            Dataset<Row> validRecords
+    ) throws DataStorageException;
+
+    public abstract void writeInvalidRecords(
+            SparkSession spark,
+            String destinationPath,
+            Dataset<Row> invalidRecords
+    ) throws DataStorageException;
 
-    @Inject
-    public StructuredZoneCDC(
-            JobArguments jobArguments,
+    // static common methods
+    protected static void appendDistinctRecords(
```

Comment:
Very minor - I'd have just implemented these as protected unless there's a reason I'm missing for these to be static?

Reply:
The static is for performance reasons.